include/aws/common/byte_buf.h

```diff
@@ -684,15 +684,20 @@ AWS_STATIC_IMPL struct aws_byte_cursor aws_byte_cursor_advance_nospec(
  * cursor unchanged.
  */
 AWS_STATIC_IMPL bool aws_byte_cursor_read(
-    struct aws_byte_cursor *AWS_RESTRICT cur,
-    void *AWS_RESTRICT dest,
+    struct aws_byte_cursor *const AWS_RESTRICT cur,
```

Comment:
Is this change necessary?  I did an informal, non-binding poll of some of the other devs here and no one really cares for this particular aspect of const (the address itself as opposed to the contents at the address) since it confuses the kind of const people normally care about and people tend to leave parameters alone anyways.  Is this useful for the prover?

Reply:
This is not required for any proofs, I just added to make it clear this function doesn't update the pointer. As the devs agreed this aspect it's not really relevant, I'll remove from this PR and I'll keep it in mind for the next ones.